src/librustdoc/visit_ast.rs

```diff
@@ -32,11 +32,28 @@ pub(crate) struct Module<'hir> {
     pub(crate) def_id: LocalDefId,
     pub(crate) renamed: Option<Symbol>,
     pub(crate) import_id: Option<LocalDefId>,
-    /// The key is the item `ItemId` and the value is: (item, renamed, import_id).
+    /// The key is the item `ItemId` and the value is: (item, renamed, Vec<import_id>).
     /// We use `FxIndexMap` to keep the insert order.
+    ///
+    /// `import_id` needs to be a `Vec` because we live in a dark world where you can have code
+    /// like:
+    ///
+    /// ```
+    /// mod raw {
+    ///     pub fn foo() {}
+    /// }
+    ///
+    /// /// Foobar
+    /// pub use raw::foo;
+    ///
+    /// pub use raw::*;
+    /// ```
+    ///
+    /// So in this case, we don't want to have two items but just one with attributes from both
+    /// imports to be merged.
```

Comment:
is merging the attributes correct? i thought that [per the reference](https://doc.rust-lang.org/reference/items/use-declarations.html#r-items.use.glob.shadowing), only items not shadowed by non-glob imports would actually be shadowed.  i guess that doesn't exactly say what happens when you shadow a glob import with an identical item...

not that i see putting docs on glob imports to be a common usecase, i think much more common is docs on the regular import and no docs on the glob import.

Reply:
Oops, the comment is wrong: we never merge attributes from glob reexports.